### PR TITLE
Move `handlers/pg/pjson` to `handlers/sjson`

### LINKED ISSUE #2495: Move `handlers/pg/pjson` to `handlers/sjson`
### What should be done?

It makes sense for PostgreSQL, SQLite, and other future SQL-based databases with no idiosyncratic JSON support to share how BSON documents are stored. Let's move `pjson` out of `pg` handler and rename it. Let's also update all documentation and comments (and fix package documentation formatting where tabs and spaces are mixed). Let's not do other refactorings because moving and refactoring a lot of code in the same issue/PR is typically hard to review.

### Where?

* https://github.com/FerretDB/FerretDB/tree/main/internal/handlers/pg/pjson
* https://github.com/FerretDB/FerretDB/tree/main/internal/handlers
* search for all references to "pjson"

### Definition of Done

- code moved;
- all comments and docs updated.

### PR DESCRIPTION
**Pull request by craigpastro**
# Description

Closes #2495. A part of #2387. Excited for SQLite ❤️ 

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I ensured that PR title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — DISMISSED** [review]
Thank you for your contribution! That change will make us closer to SQLite backend support!

File: .golangci.yml [hunk 1]
```diff
@@ -167,11 +167,6 @@ issues:
       path: cmd/envtool
       text: pgdb
 
-    # only `pg` handler can import `pjson` package, no other handler can do that
```
Comment (rumyantseva) [inline .golangci.yml:170]: In principle, we can use the same rule for `sjon`: only `pg` and future `sqlite` handlers will use `sjson`. For now, we can keep the rule for `pg`, and when `sqlite` handler is ready, we can add a similar rule for it.

See also line 15 in this file, that's one more place mentioning `pjson`.

**Review by rumyantseva — APPROVED** [review]
Thanks a lot for your contribution! Very neat approach to identifying all the `pjson` places!

I left one minor comment, but overall the PR looks good to me!

File: .golangci.yml (hunk 1 above)
Comment (craigpastro) [reply .golangci.yml:170]: Good point. Updated! Thanks. Also fixed the reference to `pjson`.

**Review by w84thesun — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**